sympy/physics/continuum_mechanics/beam.py

```diff
@@ -2667,3 +2667,104 @@ def plot_deflection(self, dir="all", subs=None):
             Py = self._plot_deflection('y')
             Pz = self._plot_deflection('z')
             return PlotGrid(3, 1, Px, Py, Pz)
+
+    def plot_loading_results(self, dir=None, subs=None):
+
+        """
+
+        Returns a subplot of Shear Force, Bending Moment,
+        Slope and Deflection of the Beam object along the direction specified.
+
+        Parameters
+        ==========
+
+        dir : string
+               Direction along which plots are required.
+               It is necessary to specify the direction for this function.
+        subs : dictionary
+               Python dictionary containing Symbols as key and their
+               corresponding values.
+
+        Examples
+        ========
+        There is a beam of length 20 meters. It it supported by rollers
+        at of its end. A linear load having slope equal to 12 is applied
+        along y-axis. A constant distributed load of magnitude 15 N is
+        applied from start till its end along z-axis.
+
+        .. plot::
+            :context: close-figs
+            :format: doctest
+            :include-source: True
+
+            >>> from sympy.physics.continuum_mechanics.beam import Beam3D
+            >>> from sympy import symbols
+            >>> l, E, G, I, A, x = symbols('l, E, G, I, A, x')
+            >>> b = Beam3D(20, 40, 21, 100, 25, x)
+            >>> b.apply_load(15, start=0, order=0, dir="z")
+            >>> b.apply_load(12*x, start=0, order=0, dir="y")
+            >>> b.bc_deflection = [(0, [0, 0, 0]), (20, [0, 0, 0])]
+            >>> R1, R2, R3, R4 = symbols('R1, R2, R3, R4')
+            >>> b.apply_load(R1, start=0, order=-1, dir="z")
+            >>> b.apply_load(R2, start=20, order=-1, dir="z")
+            >>> b.apply_load(R3, start=0, order=-1, dir="y")
+            >>> b.apply_load(R4, start=20, order=-1, dir="y")
+            >>> b.solve_for_reaction_loads(R1, R2, R3, R4)
+            >>> b.solve_slope_deflection()
+            >>> b.plot_loading_results('y')
+            PlotGrid object containing:
+            Plot[0]:Plot object containing:
+            [0]: cartesian line: -6*x**2 for x over (0.0, 20.0)
+            Plot[1]:Plot object containing:
+            [0]: cartesian line: -15*x**2/2 for x over (0.0, 20.0)
+            Plot[2]:Plot object containing:
+            [0]: cartesian line: -x**3/1600 + 3*x**2/160 - x/8 for x over (0.0, 20.0)
+            Plot[3]:Plot object containing:
+            [0]: cartesian line: x**5/40000 - 4013*x**3/90300 + 26*x**2/43 + 1520*x/903 for x over (0.0, 20.0)
+
+        """
+
+        if dir == None:
+            raise ValueError('Direction was not passed.')
+
+        dir = dir.lower();
+        deflection = self.deflection()
+
+        if dir == 'x':
+            dir_num = 0
+
+        elif dir == 'y':
+            dir_num = 1
+
+        elif dir == 'z':
+            dir_num = 2
+
+        if subs is None:
+            subs = {}
+
+        for sym in deflection[dir_num].atoms(Symbol):
```

Comment:
I think this should be done for slope, shear force, and bending moment also. Better to directly use the helper functions of the individual plotting functions (`_plot_slope() ..etc`)

Reply:
I did it for deflection only because deflection covers all the variables needed for calculating slope, bending moment, and shear force. If deflection is determinable, we can definitely determine other equations as the code works in order.

I didn't use the helper functions because I wanted to impart different colors to different functions. Using helper functions will plot all the x graphs as red, y as blue, etc unless we modify the helper functions to take color as a parameter.

Reply:
I think the idea is to have different line colors within a single plot. So you can change the color of the helper functions in a way that when you use them here all the line colors in this plot are different. (I won't recommend using color as a parameter)

Reply:
> ....because deflection covers all the variables needed for calculating slope, bending moment, and shear force

IMO there can be a use case where this won't hold true. Or is this something proven?

Reply:
> IMO there can be a use case where this won't hold true. Or is this something proven?

The functions are calculated by integrating the lower functions starting from the load. Shear Force is calculated by integrating load, Bending Moment is calculated by integrating shear force, and so on. Deflection is the function at the top so it covers everything below it and adds on some additional variables. Hence it contains all the variables needed for all the other functions. If Deflection can be calculated, it can be proven that other functions have already been calculated and the variables they contain are already present. 

Even in beam2d class, only Deflection is used for ```plot_loading_results()``` because of the same reason.